src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java

```diff
@@ -50,25 +54,48 @@ public SortClause deserialize(JsonParser parser, DeserializationContext ctxt)
     List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
     while (fieldIter.hasNext()) {
       Map.Entry<String, JsonNode> inner = fieldIter.next();
-
-      if (!inner.getValue().isInt()
-          || !(inner.getValue().intValue() == 1 || inner.getValue().intValue() == -1)) {
-        throw new JsonMappingException(
-            parser, "Sort ordering value can only be `1` for ascending or `-1` for descending.");
-      }
-
       String path = inner.getKey().trim();
-      if (path.isBlank()) {
-        throw new JsonMappingException(
-            parser, "Sort clause path must be represented as not-blank strings.");
-      }
+      if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(path)) {
+        ArrayNode arrayNode = (ArrayNode) inner.getValue();
+        float[] arrayVals = new float[arrayNode.size()];
+        if (!inner.getValue().isArray()) {
+          throw new JsonApiException(
+              ErrorCode.SHRED_BAD_VECTOR_VALUE, ErrorCode.SHRED_BAD_VECTOR_VALUE.getMessage());
+        } else {
+          if (arrayNode.size() == 0) {
+            throw new JsonApiException(
+                ErrorCode.SHRED_BAD_VECTOR_SIZE, ErrorCode.SHRED_BAD_VECTOR_SIZE.getMessage());
+          }
+          for (int i = 0; i < arrayNode.size(); i++) {
+            JsonNode element = arrayNode.get(i);
+            if (!element.isNumber()) {
+              throw new JsonApiException(
+                  ErrorCode.SHRED_BAD_VECTOR_VALUE, ErrorCode.SHRED_BAD_VECTOR_VALUE.getMessage());
+            }
+            arrayVals[i] = element.floatValue();
+          }
+        }
+        SortExpression exp = SortExpression.vsearch(arrayVals);
+        sortExpressions.clear();
```

Comment:
So I think this means that we have vector-search order by, anything else passed will be ignored?
I think this is Cassandra limitation so has to be done, but would it rather fail in such cases instead of hiding the problem?

Also: in case we get sort-by-$vector first, then other types, it seems that we would not ignore other cases?
(so we would end up with vector-order as first, then one or more "regular" order by clauses.

Reply:
Added validation